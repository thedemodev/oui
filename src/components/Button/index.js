import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Spinner from '../Spinner';

/**
 * @param {Object} props - Properties passed to component
 * @returns {ReactElement}
 */

const Button = ({
  ariaLabel,
  isSubmit,
  isLink,
  children,
  isActive,
  isDisabled,
  isLoading,
  loadingText,
  onBlur,
  onClick,
  onMouseDown,
  size,
  style,
  testSection,
  width = 'default',
  buttonRef,
  title,
}) => {

  const buttonClassNames = classNames(
    'oui-button', {
      [`oui-button--${style}`]: style,
      [`oui-button--${size}`]: size,
      [`oui-button--${width}`]: width,
      'is-active': isActive,
      'oui-button--loading': isLoading,
    });

  const type = isSubmit ? 'submit' : 'button';

  function handleOnClick(event) {
    if (isDisabled || isLoading) {
      return;
    }
    onClick(event);
  }

  if (isLink) {
    return (
      <div
        data-oui-component={ true }
        className={ buttonClassNames }
        disabled={ isDisabled }
        onBlur={ onBlur }
        data-test-section={ testSection }
        ref={ buttonRef }>
        { children }
      </div>
    );
  }

  return (
    <button
      data-oui-component={ true }
      className={ buttonClassNames }
      disabled={ isDisabled || isLoading }
      type={ type }
      onBlur={ onBlur }
      onClick={ handleOnClick }
      onMouseDown={ onMouseDown }
      data-test-section={ testSection }
      aria-label={ ariaLabel }
      aria-live="polite"
      title={ title }
      ref={ buttonRef }>
      { isLoading && <Spinner data-test-section="button-spinner" size="tiny"/> }
      { isLoading ? loadingText || 'Processing' : children }
    </button>
  );
};

Button.propTypes = {
  /** Describes buttons that have an icon but no text */
  ariaLabel: PropTypes.string,
  /** React ref to the underlying button component */
  buttonRef: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.shape({ current: PropTypes.instanceOf(Element) }),
  ]),
  /** Text within the button */
  children: PropTypes.node.isRequired,
  /** Render button with active state */
  isActive: PropTypes.bool,
  /** Prevent users from interacting with the button */
  isDisabled: PropTypes.bool,
  /** Changes the button to a div for insertion within a Link component */
  isLink: PropTypes.bool,
  /** When true, adds a spinner to the button and disables the button */
  isLoading: PropTypes.bool,
  /** Make the button act as a submit button */
  isSubmit: PropTypes.bool,
  /** When the button adds a spinner, it displays this text */
  loadingText: PropTypes.string,
  /** Function that fires when the button loses focus */
  onBlur: PropTypes.func,
  /** Function that fires when the button is clicked on */
  onClick: PropTypes.func,
  /** Function that fires when the button is mouse downed */
  onMouseDown: PropTypes.func,
  /** Various height and width options */
  size: PropTypes.oneOf([
    'tiny',
    'small',
    'large',
    'narrow',
    'tight',
  ]),
  /** Various color options */
  style: PropTypes.oneOf([
    'highlight',
    'danger',
    'danger-outline',
    'outline',
    'outline-reverse',
    'plain',
    'toggle',
    'underline',
    'unstyled',
  ]),
  /** Hook for automated JavaScript tests */
  testSection: PropTypes.string,
  /** Title of the button shown as tooltip text when the button is hovered */
  title: PropTypes.string,
  /** Various height and width options */
  width: PropTypes.oneOf([
    'default',
    'full',
  ]),
};

Button.defaultProps = {
  isLink: false,
  isLoading: false,
  isDisabled: false,
  isSubmit: false,
  loadingText: '',
  onBlur: () => {},
  onClick: () => {},
  onMouseDown: () => {},
  width: 'default',
};

Button.displayName = 'Button';

export default Button;
