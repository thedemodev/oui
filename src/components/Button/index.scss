////
/// Buttons
/// @example[html] Button colors
///   <button class="oui-button">Button</button>
///   <button class="oui-button oui-button--highlight">Button</button>
///   <button class="oui-button oui-button--danger">Button</button>
///   <button class="oui-button oui-button--danger-outline">Button</button>
///   <button class="oui-button oui-button--outline">Button</button>
///   <button class="oui-button oui-button--underline">Button <span class="oui-arrow-inline--down"></span></button>
///   <button class="oui-button oui-button--plain">Button</button>
/// @example[html] Tiny buttons
///   <button class="oui-button oui-button--tiny">Button</button>
///   <button class="oui-button oui-button--small">Button</button>
/// @example[html] Full width buttons
///   <button class="oui-button oui-button--full">Button</button>
/// @example[html] Disabled buttons
///   <button class="oui-button oui-button--disabled">Disabled with Class</button>
///   <button class="oui-button" disabled>Disabled with Attribute</button>
////

%#{$namespace}button,
.#{$namespace}button,
%oui-button,
.oui-button {
  display: inline-block;
  vertical-align: middle;
  white-space: nowrap;
  font-family: inherit;
  cursor: pointer;
  margin: 0;
  line-height: map-fetch($button, size, base, line-height);
  border-width: 1px;
  border-style: solid;
  font-size: map-fetch($button, size, base, font-size);
  font-weight: map-fetch($button, size, base, font-weight);
  user-select: none;
  border-radius: map-fetch($button, size, base, border-radius);
  height: map-fetch($button, size, base, height);
  padding: map-fetch($button, size, base, padding);
  transition: color map-fetch($transition-duration, base), background-color map-fetch($transition-duration, base), box-shadow map-fetch($transition-duration, base), border-color map-fetch($transition-duration, base);

  // Unique styles for the base, button start here.
  background-color: map-fetch($button, type, base, background);
  border-color: map-fetch($button, type, base, border);
  color: map-fetch($button, type, base, color);

  &:visited {
    color: map-fetch($button, type, base, color);
  }

  &:hover {
    color: map-fetch($button, type, base, color);
    background-color: $base-white;
  }

  &:focus {
    // Uses the same style as input hover and removes the outline while
    // remaining accessible.
    outline: none;
    border: 1px solid map-fetch($color, ui, medium);
  }

  &.is-active,
  &:active {
    background: map-fetch($button, type, base, active, background);
    border-color: map-fetch($button, type, base, active, border);
    color: map-fetch($button, type, base, active, color);
    box-shadow: map-fetch($button, type, base, active, box-shadow);
  }

  &--highlight {
    background-color: map-fetch($button, type, highlight, background);
    border-color: map-fetch($button, type, highlight, border);
    color: map-fetch($color, text, white);

    &:hover {
      background: map-fetch($button, type, highlight, hover, background);
      border-color: map-fetch($button, type, highlight, hover, border);
    }

    &:focus {
      border: 1px solid map-fetch($button, type, highlight, focus, border);
    }

    &:active {
      background: map-fetch($button, type, highlight, active, background);
      border-color: map-fetch($button, type, highlight, active, border);
      box-shadow: map-fetch($button, type, highlight, active, box-shadow);
    }

    &:visited,
    &:hover,
    &:active,
    &:focus {
      color: map-fetch($color, text, white);
    }
  }

  &--outline {
    background-color: map-fetch($button, type, outline, background);
    border-color: map-fetch($button, type, outline, border);
    color: map-fetch($button, type, outline, color);

    &:visited {
      color: map-fetch($button, type, outline, color);
    }

    &:focus {
      border: 1px solid map-fetch($button, type, outline, hover, border);
    }

    &:hover {
      background-color: map-fetch($button, type, outline, background);
      border-color: map-fetch($button, type, outline, hover, border);
      color: map-fetch($button, type, outline, hover, color);
      box-shadow: none;
    }

    &:active {
      background-color: map-fetch($button, type, outline, background);
      border-color: map-fetch($button, type, outline, active, border);
      color: map-fetch($button, type, outline, active, color);
      box-shadow: none;
    }
  }

  &--danger {
    background-color: map-fetch($button, type, danger, background);
    border-color: map-fetch($button, type, danger, border);
    color: map-fetch($color, text, white);

    &:hover {
      background: map-fetch($button, type, danger, hover, background);
      border-color: map-fetch($button, type, danger, hover, border);
    }

    &:active {
      background: map-fetch($button, type, danger, active, background);
      border-color: map-fetch($button, type, danger, active, border);
    }

    &:focus {
      border: 1px solid shade(map-fetch($button, type, danger, background), 30%);
    }

    &:visited,
    &:hover,
    &:active,
    &:focus {
      color: map-fetch($color, text, white);
    }
  }

  &--danger-outline {
    color: map-fetch($button, type, danger-outline, color);
    border-color: map-fetch($button, type, danger-outline, border);
    background-color: transparent;
    box-shadow: none;

    &:focus {
      border: 1px solid shade(map-fetch($button, type, danger, background), 30%);
    }

    &:hover {
      color: map-fetch($button, type, danger-outline, hover, color);
      border-color: map-fetch($button, type, danger-outline, hover, border);
      background-color: transparent;
      box-shadow: none;
    }

    &:active {
      border-color: map-fetch($button, type, danger-outline, active, border);
      background-color: transparent;
      color: map-fetch($button, type, danger-outline, active, color);
      box-shadow: none;
    }
  }

  &--plain,
  &--underline {
    color: map-fetch($button, type, plain, color);
    border-color: transparent;
    background-color: transparent;
    box-shadow: none;

    &:hover {
      color: map-fetch($button, type, plain, hover, color);
    }

    &:active {
      color: map-fetch($button, type, plain, active, color);
    }

    &:visited,
    &:hover,
    &:active,
    &:focus {
      border-color: transparent;
      background-color: transparent;
      box-shadow: none;
    }
  }

  &--underline {
    border-radius: 0 !important;
  }

  &--plain {
    &:focus, &:hover {
      border: 1px solid var(--light-blue-25);
    }
  }

  &--toggle {
    &:hover {
      color: map-fetch($button, type, base, color);
    }

    &:active,
    &.is-active:hover {
      color: map-fetch($button, type, base, active, color);
    }

    &:not(:hover):not(:active):not(.is-active) {
      border-color: transparent;
      background-color: transparent;
    }
  }

  &--underline {
    border-bottom: 2px solid map-fetch($button, type, underline, border);
    color: map-fetch($button, type, base, color);
    padding-left: spacer(0.5);
    padding-right: spacer(0.5);

    &:hover {
      border-bottom-width: 2px;
      border-bottom-color: map-fetch($button, type, underline, hover, border);
      color: map-fetch($button, type, underline, color);
      & .oui-arrow-inline--down {
        color: map-fetch($button, type, underline, hover, arrow-color);
      }
    }

    &:focus {
      border-bottom-width: 2px;
      border-bottom-color: map-fetch($button, type, underline, focus, border);
      & .oui-arrow-inline--down {
        color: map-fetch($button, type, underline, focus, border);
      }
    }

    &:active {
      border-bottom-width: 2px;
      border-bottom-color: map-fetch($button, type, underline, active, border);
      color: map-fetch($button, type, underline, active, color);
      & .oui-arrow-inline--down {
        color: map-fetch($button, type, underline, active, arrow-color);
      }
    }
  }

  // Button Sizes

  &--tiny {
    font-size: map-fetch($button, size, tiny, font-size);
    font-weight: map-fetch($button, size, tiny, font-weight);
    border-radius: map-fetch($button, size, tiny, border-radius);
    line-height: map-fetch($button, size, tiny, line-height);
    height: map-fetch($button, size, tiny, height);
    padding: map-fetch($button, size, tiny, padding);
  }

  &--small {
    font-size: map-fetch($button, size, small, font-size);
    font-weight: map-fetch($button, size, small, font-weight);
    border-radius: map-fetch($button, size, small, border-radius);
    line-height: map-fetch($button, size, small, line-height);
    height: map-fetch($button, size, small, height);
    padding: map-fetch($button, size, small, padding);
  }

  &--large {
    font-size: map-fetch($button, size, large, font-size);
    font-weight: map-fetch($button, size, large, font-weight);
    border-radius: map-fetch($button, size, large, border-radius);
    line-height: map-fetch($button, size, large, line-height);
    height: map-fetch($button, size, large, height);
    padding: map-fetch($button, size, large, padding);
  }

  &--narrow {
    padding-left: spacer(1);
    padding-right: spacer(1);
  }

  &--tight {
    padding-left: spacer(0.5);
    padding-right: spacer(0.5);
  }

  &--full {
    width: 100%;
    text-align: center;
  }

  // With Icon
  // Some extra help is needed to get the icon and text to line up.

  &--icon {
    line-height: 1.4;

    .#{$namespace}icon {
      vertical-align: sub;
    }
  }

  &--unstyled {
    // scss-lint:disable ImportantRule
    background: inherit !important;
    border: none !important;
    border-radius: inherit !important;
    box-shadow: inherit !important;
    color: inherit !important;
    font-size: inherit !important;
    font-weight: inherit !important;
    height: inherit !important;
    line-height: 1 !important;
    margin: 0 !important;
    padding: 0 !important;
    text-shadow: inherit !important;
  }

  // Disabled State
  // Used either an attribute or a class, depending on the use case.

  &[disabled],
  &--disabled,
  &--disabled:visited,
  &--disabled:hover,
  &--disabled:active,
  &--disabled:focus {
    cursor: default;
    box-shadow: none;
    background-color: map-fetch($button, type, disabled, background);
    border-color: map-fetch($button, type, disabled, border);
    color: map-fetch($button, type, disabled, color);
    box-shadow: none;

    &.#{$namespace}button--plain,
    &.#{$namespace}button--toggle,
    &.oui-button--plain,
    &.oui-button--toggle {
      background-color: transparent;
      border-color: transparent;
    }

    &.#{$namespace}button--outline-reverse,
    &.oui-button--outline-reverse {
      background-color: transparent;
    }
  }

  &--loading {
    .oui-spinner {
      margin-bottom: spacer(-0.5);
      margin-right: spacer(0.5);
    }
  }
}
