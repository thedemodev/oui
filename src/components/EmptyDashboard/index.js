import React from 'react';
import PropTypes from 'prop-types';

const EmptyDashboard = ({
  button,
  imageHeight,
  imagePath,
  imageWidth,
  headline,
  description,
  descriptionMaxWidth,
  showButtonBelow,
  testSection,
}) => {
  return (
    <div
      data-oui-component={ true }
      className='flex flex--column overflow-y--scroll flex--1'
      data-test-section='layers-empty-state'>
      { button && !showButtonBelow && (
        <div className="flex push-quad--sides oui-empty-dashboard__button-row flex-justified--end">
          { button }
        </div>
      ) }
      <div className='anchor--middle height--1-1 flex flex--row flex-align--center'>

        <div
          data-oui-component={ true }
          className='text--center soft-quad'
          data-test-section={ testSection }>

          { imagePath && (
            <div className="push-double--bottom">
              <img
                src={ imagePath }
                className="svg--non-scaling-stroke display--inline"
                style={{
                  maxWidth: '450px',
                  maxHeight: '290px',
                  height: imageHeight,
                  width: imageWidth,
                }}
                data-test-section={ testSection && `${testSection}-image` }
                alt=""
              />
            </div>
          ) }

          { headline && (
            <h1
              className="alpha push--bottom"
              data-test-section={ testSection && `${testSection}-headline` }>
              { headline }
            </h1>
          ) }

          { description && (
            <div
              className="muted push--bottom anchor--middle"
              style={{ maxWidth: descriptionMaxWidth }}
              data-test-section={ testSection && `${testSection}-description` }>
              { description }
            </div>
          ) }

          { button && showButtonBelow && (
            <div className="push-double--top">
              { button }
            </div>
          ) }
        </div>

      </div>
    </div>
  );
};

EmptyDashboard.propTypes = {
  /** A button, link, or other element that is actionable */
  button: PropTypes.node,
  /** A string or `div` containing longer text that describes the empty state */
  description: PropTypes.element,
  /** Max width of description. Accepts any valid CSS width value. */
  descriptionMaxWidth: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  /** Short text about the empty state */
  headline: PropTypes.string.isRequired,
  /** String to set the max image height */
  imageHeight: PropTypes.string,
  /** Path to an image representing the empty state */
  imagePath: PropTypes.string,
  /** String to set the max image width */
  imageWidth: PropTypes.string,
  /**
   * Boolean used to show the button below the image
   * instead of top right corner of the empty state
   */
  showButtonBelow: PropTypes.bool,
  /** Identifier used to create data-test-section attributes for testing */
  testSection: PropTypes.string,
};

EmptyDashboard.getDefaultProps = {
  descriptionMaxWidth: '450px',
  imageHeight: 'inherit',
  imageWidth: 'inherit',
  showButtonBelow: false,
  testSection: 'empty-hello',
};

export default EmptyDashboard;
