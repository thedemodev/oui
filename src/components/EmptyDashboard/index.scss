.oui-empty-dashboard__button-row {
  margin-top: 20px;
  margin-bottom: -40px;
  z-index: 1;
}
