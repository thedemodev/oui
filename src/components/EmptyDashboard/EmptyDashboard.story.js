import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs, text, select } from '@storybook/addon-knobs';

import Button from '../Button/';
import Link from '../Link/';
import EmptyDashboard from './index.js';

const stories = storiesOf('Empty Dashboard', module);
stories
  .addDecorator(withKnobs)
  .addDecorator(story => (
    <div id="root-preview">
      {story()}
    </div>
  ));

stories
  .add('Default empty', () => {
    return (
      <EmptyDashboard
        headline={ text('headline', 'Get started with Optimizely X') }
        description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> }
        button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
        imagePath={ select('imagePath', {
          'Campaigns': 'https://app.optimizely.com/static/img/x/empty-states/campaign-list-empty-state.svg',
          'Extensions': 'https://app.optimizely.com/static/img/x/empty-states/extensions-empty-state.svg',
          'SST': 'https://app.optimizely.com/static/img/x/empty-states/custom-project-experiments-empty-state.svg',
          'Rollouts': 'https://app.optimizely.com/static/img/x/empty-states/empty-rollouts.svg',
          'Feature Flags': 'https://app.optimizely.com/static/img/x/empty-states/empty-feature-flags.svg',
          'Audiences': 'https://app.optimizely.com/static/img/x/empty-states/audience-list-empty-state.svg',
          'Attribute': 'https://app.optimizely.com/static/img/x/empty-states/attribute-dashboard-empty-state.svg',
          'Page List': 'https://app.optimizely.com/static/img/x/empty-states/page-list-empty-state.svg',
        }, 'https://app.optimizely.com/static/img/x/empty-states/campaign-list-empty-state.svg') }
      />
    );
  })
  .add('All images', () => {
    return (
      <div>
        <EmptyDashboard
          headline={ 'Get started with Campaigns' }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> }
          button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath={ 'https://app.optimizely.com/static/img/x/empty-states/campaign-list-empty-state.svg' }
        />
        <hr/>
        <EmptyDashboard
          headline={ 'Get started with Extensions' }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> }
          button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath={ 'https://app.optimizely.com/static/img/x/empty-states/extensions-empty-state.svg' }
        />
        <hr/>
        <EmptyDashboard
          headline={ 'Get started with Experiments' }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> }
          button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath={ 'https://app.optimizely.com/static/img/x/empty-states/custom-project-experiments-empty-state.svg' }
        />
        <hr/>
        <EmptyDashboard
          headline={ 'Get started with Rollouts' }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> }
          button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath={ 'https://app.optimizely.com/static/img/x/empty-states/empty-rollouts.svg' }
        />
        <hr/>
        <EmptyDashboard
          headline={ 'Get started with Feature Flags' }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> } button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath={ 'https://app.optimizely.com/static/img/x/empty-states/empty-feature-flags.svg' }
        />
        <hr/>
        <EmptyDashboard
          headline={ 'Get started with Audiences' }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> } button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath={ 'https://app.optimizely.com/static/img/x/empty-states/audience-list-empty-state.svg' }
        />
        <hr/>
        <EmptyDashboard
          headline={ 'Get started with Attributes' }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> } button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath={ 'https://app.optimizely.com/static/img/x/empty-states/attribute-dashboard-empty-state.svg' }
        />
        <hr/>
        <EmptyDashboard
          headline={ 'Get started with Pages' }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> } button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath={ 'https://app.optimizely.com/static/img/x/empty-states/page-list-empty-state.svg' }
        />
        <hr/>
        <EmptyDashboard
          headline={ 'Get started with Change History' }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> } button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath='https://app.optimizely.com/static/img/x/empty-states/change-history-empty-state.svg'
        />
        <hr/>
        <EmptyDashboard
          headline={ 'Get started with Campaign Manager' }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> } button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath='https://app.optimizely.com/static/img/x/empty-states/campaign-manager-empty.svg'
        />
        <hr/>
        <EmptyDashboard
          headline={ 'Get started with Custom Snippets' }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> } button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath='https://app.optimizely.com/static/img/x/empty-states/custom-snippets-empty-state.svg'
        />
        <hr/>
        <EmptyDashboard
          headline={ 'Get started with Fullstack Variables' }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> } button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath='https://app.optimizely.com/static/img/x/empty-states/fullstack-variables-empty-state.svg'
        />
        <hr/>
        <EmptyDashboard
          headline={ 'Get started with Integrations' }
          description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> } button={ (<Button style="highlight">{ text('buttonText', 'Highlight') }</Button>) }
          imagePath='https://app.optimizely.com/static/img/x/empty-states/integrations-empty.svg'
        />
      </div>
    );
  });
