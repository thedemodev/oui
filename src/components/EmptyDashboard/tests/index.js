import React from 'react';
import Button from '../../Button/';
import EmptyDashboard from '../index.js';
import Link from '../../Link/';
import { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

describe('components/EmptyDashboard', () => {
  it('should render', () => {
    const output = mount(
      <EmptyDashboard
        headline='Get started with Optimizely X'
        description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> }
        button={ (<Button style="highlight">Highlight Button</Button>) }
        imagePath='https://app.optimizely.com/static/img/p13n/campaign-list-empty-state.svg'
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should render with button below image when showButtonBelow property is set to TRUE', () => {
    const output = mount(
      <EmptyDashboard
        headline='Get started with Optimizely X'
        description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> }
        button={ (<Button style="highlight">Highlight Button</Button>) }
        imagePath='https://app.optimizely.com/static/img/p13n/campaign-list-empty-state.svg'
        showButtonBelow={ true }
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should render with specified height and width when passed using the imageHeight and imageWidth properties', () => {
    const output = mount(
      <EmptyDashboard
        headline='Get started with Optimizely X'
        description={ <div>This is a test description about <Link href='http://google.com'>X Feature</Link> with a Google.</div> }
        button={ (<Button style="highlight">Highlight Button</Button>) }
        imagePath='https://app.optimizely.com/static/img/p13n/campaign-list-empty-state.svg'
        imageHeight='300px'
        imageWidth='200px'
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
