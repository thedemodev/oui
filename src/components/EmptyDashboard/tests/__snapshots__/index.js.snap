// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/EmptyDashboard should render 1`] = `
<EmptyDashboard
  button={
    <Button
      isDisabled={false}
      isLink={false}
      isLoading={false}
      isSubmit={false}
      loadingText=""
      onBlur={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
      style="highlight"
      width="default"
    >
      Highlight Button
    </Button>
  }
  description={
    <div>
      This is a test description about 
      <Link
        href="http://google.com"
      >
        X Feature
      </Link>
       with a Google.
    </div>
  }
  headline="Get started with Optimizely X"
  imagePath="https://app.optimizely.com/static/img/p13n/campaign-list-empty-state.svg"
>
  <div
    className="flex flex--column overflow-y--scroll flex--1"
    data-oui-component={true}
    data-test-section="layers-empty-state"
  >
    <div
      className="flex push-quad--sides oui-empty-dashboard__button-row flex-justified--end"
    >
      <Button
        isDisabled={false}
        isLink={false}
        isLoading={false}
        isSubmit={false}
        loadingText=""
        onBlur={[Function]}
        onClick={[Function]}
        onMouseDown={[Function]}
        style="highlight"
        width="default"
      >
        <button
          aria-live="polite"
          className="oui-button oui-button--highlight oui-button--default"
          data-oui-component={true}
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
        >
          Highlight Button
        </button>
      </Button>
    </div>
    <div
      className="anchor--middle height--1-1 flex flex--row flex-align--center"
    >
      <div
        className="text--center soft-quad"
        data-oui-component={true}
      >
        <div
          className="push-double--bottom"
        >
          <img
            alt=""
            className="svg--non-scaling-stroke display--inline"
            src="https://app.optimizely.com/static/img/p13n/campaign-list-empty-state.svg"
            style={
              Object {
                "height": undefined,
                "maxHeight": "290px",
                "maxWidth": "450px",
                "width": undefined,
              }
            }
          />
        </div>
        <h1
          className="alpha push--bottom"
        >
          Get started with Optimizely X
        </h1>
        <div
          className="muted push--bottom anchor--middle"
          style={
            Object {
              "maxWidth": undefined,
            }
          }
        >
          <div>
            This is a test description about 
            <Link
              href="http://google.com"
            >
              <a
                className="link"
                data-oui-component={true}
                disabled={false}
                href="http://google.com"
              >
                X Feature
              </a>
            </Link>
             with a Google.
          </div>
        </div>
      </div>
    </div>
  </div>
</EmptyDashboard>
`;

exports[`components/EmptyDashboard should render with button below image when showButtonBelow property is set to TRUE 1`] = `
<EmptyDashboard
  button={
    <Button
      isDisabled={false}
      isLink={false}
      isLoading={false}
      isSubmit={false}
      loadingText=""
      onBlur={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
      style="highlight"
      width="default"
    >
      Highlight Button
    </Button>
  }
  description={
    <div>
      This is a test description about 
      <Link
        href="http://google.com"
      >
        X Feature
      </Link>
       with a Google.
    </div>
  }
  headline="Get started with Optimizely X"
  imagePath="https://app.optimizely.com/static/img/p13n/campaign-list-empty-state.svg"
  showButtonBelow={true}
>
  <div
    className="flex flex--column overflow-y--scroll flex--1"
    data-oui-component={true}
    data-test-section="layers-empty-state"
  >
    <div
      className="anchor--middle height--1-1 flex flex--row flex-align--center"
    >
      <div
        className="text--center soft-quad"
        data-oui-component={true}
      >
        <div
          className="push-double--bottom"
        >
          <img
            alt=""
            className="svg--non-scaling-stroke display--inline"
            src="https://app.optimizely.com/static/img/p13n/campaign-list-empty-state.svg"
            style={
              Object {
                "height": undefined,
                "maxHeight": "290px",
                "maxWidth": "450px",
                "width": undefined,
              }
            }
          />
        </div>
        <h1
          className="alpha push--bottom"
        >
          Get started with Optimizely X
        </h1>
        <div
          className="muted push--bottom anchor--middle"
          style={
            Object {
              "maxWidth": undefined,
            }
          }
        >
          <div>
            This is a test description about 
            <Link
              href="http://google.com"
            >
              <a
                className="link"
                data-oui-component={true}
                disabled={false}
                href="http://google.com"
              >
                X Feature
              </a>
            </Link>
             with a Google.
          </div>
        </div>
        <div
          className="push-double--top"
        >
          <Button
            isDisabled={false}
            isLink={false}
            isLoading={false}
            isSubmit={false}
            loadingText=""
            onBlur={[Function]}
            onClick={[Function]}
            onMouseDown={[Function]}
            style="highlight"
            width="default"
          >
            <button
              aria-live="polite"
              className="oui-button oui-button--highlight oui-button--default"
              data-oui-component={true}
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onMouseDown={[Function]}
              type="button"
            >
              Highlight Button
            </button>
          </Button>
        </div>
      </div>
    </div>
  </div>
</EmptyDashboard>
`;

exports[`components/EmptyDashboard should render with specified height and width when passed using the imageHeight and imageWidth properties 1`] = `
<EmptyDashboard
  button={
    <Button
      isDisabled={false}
      isLink={false}
      isLoading={false}
      isSubmit={false}
      loadingText=""
      onBlur={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
      style="highlight"
      width="default"
    >
      Highlight Button
    </Button>
  }
  description={
    <div>
      This is a test description about 
      <Link
        href="http://google.com"
      >
        X Feature
      </Link>
       with a Google.
    </div>
  }
  headline="Get started with Optimizely X"
  imageHeight="300px"
  imagePath="https://app.optimizely.com/static/img/p13n/campaign-list-empty-state.svg"
  imageWidth="200px"
>
  <div
    className="flex flex--column overflow-y--scroll flex--1"
    data-oui-component={true}
    data-test-section="layers-empty-state"
  >
    <div
      className="flex push-quad--sides oui-empty-dashboard__button-row flex-justified--end"
    >
      <Button
        isDisabled={false}
        isLink={false}
        isLoading={false}
        isSubmit={false}
        loadingText=""
        onBlur={[Function]}
        onClick={[Function]}
        onMouseDown={[Function]}
        style="highlight"
        width="default"
      >
        <button
          aria-live="polite"
          className="oui-button oui-button--highlight oui-button--default"
          data-oui-component={true}
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
        >
          Highlight Button
        </button>
      </Button>
    </div>
    <div
      className="anchor--middle height--1-1 flex flex--row flex-align--center"
    >
      <div
        className="text--center soft-quad"
        data-oui-component={true}
      >
        <div
          className="push-double--bottom"
        >
          <img
            alt=""
            className="svg--non-scaling-stroke display--inline"
            src="https://app.optimizely.com/static/img/p13n/campaign-list-empty-state.svg"
            style={
              Object {
                "height": "300px",
                "maxHeight": "290px",
                "maxWidth": "450px",
                "width": "200px",
              }
            }
          />
        </div>
        <h1
          className="alpha push--bottom"
        >
          Get started with Optimizely X
        </h1>
        <div
          className="muted push--bottom anchor--middle"
          style={
            Object {
              "maxWidth": undefined,
            }
          }
        >
          <div>
            This is a test description about 
            <Link
              href="http://google.com"
            >
              <a
                className="link"
                data-oui-component={true}
                disabled={false}
                href="http://google.com"
              >
                X Feature
              </a>
            </Link>
             with a Google.
          </div>
        </div>
      </div>
    </div>
  </div>
</EmptyDashboard>
`;
